metric_config_parser/data_source.py

```diff
@@ -127,8 +156,15 @@ class DataSourceDefinition:
     build_id_column: Optional[str] = None
     friendly_name: Optional[str] = None
     description: Optional[str] = None
+    joins: Optional[Dict[str, Dict[str, Any]]] = None
+    columns_as_dimensions: Optional[bool] = None
```

Comment:
This is needed for https://github.com/mozilla/lookml-generator/pull/969 to define whether selected fields of a data source should be exposed and show up as Looker dimensions. This could be something we might want to support in opmon too.